app/Http/Controllers/CanteensController.php

```diff
@@ -46,10 +44,41 @@ public function byWeek(Request $request)
         
         $canteens = Canteens::with('menus')->whereBetween('date', [$start,$end])->orderBy('date')->get();
         
-        if ($canteens->isEmpty()) {
+        
+        
+        return self::_after($request, $canteens);
+    }
+    
+    
+    private static function _modSince($request)
+    {
+        return new Carbon(str_replace(':', '', $request->header('If-modified-since', ': 1970-01-01')));
+    }
+    
+    private function _after($request, $result)
+    {
+        if ($result->isEmpty()) {
             return response()->json([], 404);
         }
         
-        return $canteens;
+        
+        $modSince = self::_modSince($request);
+        
+        
+        // Find max date
+        $maxDate = new Carbon('1970-01-01');
+        $result->each(function ($post) use ($maxDate) {
```

Comment:
We should use the DB to get the latest posts' updated at field

Reply:
We only need to search in the results of the query, not in the whole data set, as **we don't need to re-send `Post id=20` just because `Post id=21` was modified.** We need to find the maximum `updated_at` in the result collection, which is ordered by its `date` field, so the `updated_at` fields is considered unordered, so the best search we can do is linear, which we do.
Of course, we could run the query _twice_, which would mean we have to search a much larger database - which is way less effective. We would also lose the ability to move date checking into a common function, as the second query would need to be the same as the first, except we order by `updated_at`
The only other way I can think of is to export the result into a new table, and query that. But then it needs sorting/maximum search, so we're at where we started _at best_.